docs/references/cli.md

```diff
@@ -0,0 +1,6 @@
+# CLI commands
+
+<note to developer - this is where you put the details of each CLI command or
+other [reference](https://diataxis.fr/reference/) material for your project. 
+This declutters the user guide so you don't have to document every flavor of a
+ command/api etc>
```

Comment:
It would be preferable to put as much reference content in source as possible, especially for things like CLI commands. It's better to have manpages that you can invoke in a terminal than to have to switch context and view standalone docs.

Rather than decluttering the user guide maybe it's the fact that you're keeping that sort of content task-oriented.

Reply:
I'm not sure what you're advising here. Are you suggesting the reference content SHOULD be in the user guide?  So this is the kind of user guide page I'm trying to avoid - https://docs.ansible.com/ansible-core/devel/collections_guide/collections_installing.html

I also disagree that users should only depend on manpages at a terminal if that was the suggestion? We should have solid cli references, config references, etc.

Reply:
ok clarified in matrix - The advice here is to change the note to add something like: It's best if you can generate this from the source code so man pages and docs match.

Basically we want to encourage projects to generate reference material from the source code as much as possible as that is more easily maintained and keeps things like docs and man pages in sync.  The other idea to add to the toolkit is to consider creating man pages in .md format so they can be single-sourced to the documentation as well.

Reply:
I just pushed a commit with some suggestions. please take a look.